### ISSUE TITLE
Configurable order of properties in Concept pages

### ISSUE THREAD

**Issue by tfrancart** [NatLibFi/Skosmos#554]
Some thesaurus managers would like to have the ordering of properties in the Concept page according to what ISO-25964 recommends, which I am copy-pasting below :

> 12.2.2 Single record display
> When the record for a concept or a preferred term is viewed individually the preferred term should normally be displayed first and the other fields are customarily presented after it in the following sequence:
> a) CC concept code or notation, locating the term/concept in a hierarchical sequence;
> b) SN scope note;
> c) UF references to non-preferred equivalent term(s);
> d) TT references to top terms;
> e) BT references to broader terms;
> f) NT references to narrower terms;
> g) RT references to related terms;
> h) DEF definition of the preferred term (see 6.2.3);
> i) HN history note (see 6.2.4);
> j) SC subject category or other concept group to which the concept has been assigned.
> In this arrangement, the SN and UF are given near the beginning of the list because they clarify the scope of the concept. They are followed by relationships with other terms. DEF and HN are given last because they are viewed as administrative fields, used more by the thesaurus editor than the indexer or searcher. As an option, these last fields may be shown after the scope note, bringing all text fields together. However, it is important to ensure that SN and DEF are not confused when both are present.

It is not complicated to reorder the properties of the concept by editing line 62 in Concept.php :

> $this->order = array("rdf:type", "dc:isReplacedBy", "skos:definition", "skos:broader", "skos:narrower", "skos:related", "skos:altLabel", "skosmos:memberOf", "skos:note", "skos:scopeNote", "skos:historyNote", "rdfs:comment", "dc11:source", "dc:source", "skos:prefLabel");

However this doesn't allow to control the order of the synonyms and the group information. Having the possibility to easily configure this - without tempering with the HTML template - would be nice. A configuration option, maybe, like "UseIso25964PropertyOrder", or something similar ?

Cheers

**Comment by osma** [comment]
I guess we didn't look at ISO 25964 back in the days the order was set. Before adding yet another configuration option, we could at least consider just adopting the ISO 25964 order as the default.

**Comment by tfrancart** [comment]
I uderstood you did some usability testing when designing the screen, which is also a good argument for the order as it is today, that's why I was proposing this as an option.

**Comment by osma** [comment]
Alternative would be to make the order entirely configurable (perhaps defaulting to ISO order)

**Comment by osma** [comment]
> Alternative would be to make the order entirely configurable (perhaps defaulting to ISO order)

More specifically:

In the global section in config.ttl, there could be zero or more definitions of property orders (expressed as RDF lists). Then each vocabulary could be set to follow one of these orders. If no property order would be set for some vocabulary, it would use a default order (similar to the current hardcoded order, perhaps with some tweaks to better follow ISO 25964).

**Comment by osma** [comment]
This is how the configuration could look.

ISO 25964 order:

```
:myvoc skosmos:propertyOrder skosmos:iso25964PropertyOrder .
```

Default Skosmos order:


```
:myvoc skosmos:propertyOrder skosmos:defaultPropertyOrder .
```

Custom order:

```
:myvoc skosmos:propertyOrder ( rdf:type dc:isReplacedBy skos:definition skos:broader skos:narrower skos:related skos:altLabel skosmos:memberOf skos:note skos:scopeNote skos:historyNote rdfs:comment dc11:source dc:source skos:prefLabel) .
```

**Comment by osma** [comment]
PR #992 caused a regression on the search result page:

![kuva](https://user-images.githubusercontent.com/1132830/84634184-212ee580-aefa-11ea-8dbe-8ddb634aaaf1.png)

(from: http://dev.finto.fi/yso/fi/search?clang=fi&q=kissa )

The group information is now duplicated (with the second line badly spaced) and the foreign labels are shown as "Array, Array, Array".

**Comment by osma** [comment]
Documented in the wiki: https://github.com/NatLibFi/Skosmos/wiki/Setting-the-property-order

**Comment by osma** [comment]
> The group information is now duplicated (with the second line badly spaced)

This was fixed in PR #1010

> and the foreign labels are shown as "Array, Array, Array".

This was fixed in commit 161b0cb9616410550472b20d2afef12e261cb654 directly to `master`
